ygen/protogen.go

```diff
@@ -208,7 +224,10 @@ func writeProto3Msg(msg *yangDirectory, msgs map[string]*yangDirectory, state *g
 // within the YANG schema and returns a protoMsg which can be mapped to the protobuf
 // code representing it. It uses the set of messages that have been extracted and the
 // current generator state to map to other messages and ensure uniqueness of names.
-func genProto3Msg(msg *yangDirectory, msgs map[string]*yangDirectory, state *genState, compressPaths bool) ([]protoMsg, []error) {
+// The configuration parameters for the current code generation required are supplied
+// as arguments, particularly whether path is compression is enabled, the base package
+// name and the name of the package that enumerated types are written to.
+func genProto3Msg(msg *yangDirectory, msgs map[string]*yangDirectory, state *genState, compressPaths bool, basePackageName, enumPackageName string) ([]protoMsg, []error) {
```

Comment:
List of params is quite long - suggest either a struct or changing the function comment to be in list format.